backend/python/app/tests/graphql/mutations/test_story_mutation.py

```diff
@@ -0,0 +1,163 @@
+import re
+
+from ....models.story import Story
+
+
```

Comment:
Lots of concerns here:

- Well this test fails for a type mismatch.
```
At index 0 diff: {'id': 1, 'story_id': '1', 'line_index': 0, 'cont
ent': 'line 1'} != {'id': 1, 'story_id': 1, 'line_index': 0, 'content': 'line 1'
```
Conversely, when I make the expected type into char, the dict conversion wants `line_index` as an integer
```
 {'createStory': {'ok': True, 'story': {'contents': [{'content': 'line 1', 'id': 1, 'lineIndex': 0, 'storyId': '1'}, {'...'id': 3, 'lineIndex': 2, 'storyId': '1'}]
, 'description': 'this should explain things', 'id': 1, 'level': 10001, ...}}} != {'createStory': {'ok': True, 'story': {'contents': [{'content': 'line 1', 'id': 1, 'lineIn
dex': '0', 'storyId': '1'}, ...d': 3, 'lineIndex': '2', 'storyId': '1'}], 'description': 'this should explain things', 'id': 1, 'level': 10001, ...}}}
```
- This is feels kinda hacky, particularly how I'm trying to work around the snake case conversion, and nesting of the contents array. 
- What do we actually want to test here? I've basically asserted that the db entry and response against expected. Is that right?

Reply:
### Reply from @gaoxk:

```
At index 0 diff: {'id': 1, 'story_id': '1', 'line_index': 0, 'cont
ent': 'line 1'} != {'id': 1, 'story_id': 1, 'line_index': 0, 'content': 'line 1'
```
This is happening because `value` and `test_db_obj[to_snake_case(pattern, field_name)]` are different dict objects in python's memory. [Something like this will work.](https://stackoverflow.com/questions/4527942/comparing-two-dictionaries-and-checking-how-many-key-value-pairs-are-equal)

>This is feels kinda hacky, particularly how I'm trying to work around the snake case conversion, and nesting of the contents array.

The snake case conversion is quite unfortunate. This test also has some repetition; story data and contents are explicitly written out in the gql request and the response test. Declaring the expected model first may alleviate some pains:
```
new_story = Story(title="title test", description="this should explain things", ...)
result = client.execute(
        """
         mutation CreateStory($storyData: ...){
            createStory(
                contents: $contents,
...
       variables={storyData:{new_story.title, ...}...},
)
...
assert dict(result["data"]) == {
        "createStory": {
            "ok": True,
            "story": new_story.to_dict(camel_case=True, include_relationships=True)
....
assert test_db_obj.title == new_story.title
# could be replaced with a DIY generic attribute checker, sqlalchemy doesn't have a built in
# could be replaced with assert_story_equals_model from test_story_service.py
...
```

> What do we actually want to test here? I've basically asserted that the db entry and response against expected. Is that right?

yep! The essential concept looks good to me

Reply:
From discussions -> keep snapshots ([docs](https://docs.graphene-python.org/en/latest/testing/#snapshot-testing)) as an investigation in future PR. Not sure about it, so avoid having to rebuild